### DISCUSSION TITLE
why applying RRC filter before FSK modulating and not after it?

### DISCUSSION THREAD

**Discussion by sporadicgroups** [jgaeddert/liquid-dsp#269] [Q&A] [answered]
sorry from my bad English!

In DMR standards "ETSI TS 102 361-1" 
10.2.2.3 4FSK Modulator
they use a RRC filter before frequency modulating the signal

![image](https://user-images.githubusercontent.com/93916666/141760935-8fb0c8bd-435b-48c6-93dc-e66874d9e6aa.png)

is there any reason why did they applying RRC filter before FSK modulating  and not after?

anyway I was trying to implement these two steps in LIQUID but i face some understanding problems:

I implement RRC filter part using the `firinterp_rrrf` witch take and output `float` samples by using `firinterp_rrrf_execute`

and using `fskmod`  to implement the fsk part, but since `fskmod_modulate` accepts an `integer` symbols and not `float`, i could not pass the values from the output of the RRC filter to FSK modulator

i am fresh student so be easy on me if i miss something

**Comment by jgaeddert** [comment, marked as answer]
No problem. I'm pretty sure the reason for the filter is to reduce the occupied bandwidth. Keep in mind that "frequency modulation" is the generic analog form, here, and should not be confused with "frequency shift keying". DMR is using the RRC filter as an input to the frequency modulation together to act as a continuous-phase FSK modem. This would be akin to feeding the output of `firinterp_rrrf` into `freqmod`. You'll need to be very careful of the signal level and gain going into the frequency modulator as this will have a significant impact on the distortion and occupied bandwidth of the output 4FSK signal.
